client/gutenberg/editor/controller.js

```diff
@@ -101,16 +100,18 @@ export const authenticate = ( context, next ) => {
 	const state = context.store.getState();
 
 	const siteId = getSelectedSiteId( state );
+	const isJetpack = isJetpackSite( state, siteId );
 	const isDesktop = isEnabled( 'desktop' );
 	const storageKey = `gutenframe_${ siteId }_is_authenticated`;
 
 	let isAuthenticated =
 		globalThis.sessionStorage.getItem( storageKey ) || // Previously authenticated.
-		! isSiteWpcomAtomic( state, siteId ) || // Simple sites users are always authenticated.
+		! isJetpack || // Simple sites users are always authenticated.
```

Comment:
Do we need to update this comment to clarify whether Atomic sites are authenticated?

Reply:
I've accepted the PR but maybe it worths updating this comment.

Reply:
Yes, that probably needs more explanation. The logic is still targeting simple sites though. I'll add an extra note